### PR TITLE
Fix calendar view code

### PR DESCRIPTION
**Pull request by Tolsa**
- Adding Arnaudb's code to fix calendar view (initially for Zentist).

- see https://trello.com/c/Y7MJUIKn/2598-p2-smart-view-smart-view-not-showing-as-a-calendar-in-the-related-data-section-of-a-summary-view for more info

### CODE REVIEW

File: _includes/common/views/smart-view.md [hunk 1]
```diff
@@ -471,21 +500,23 @@ export default Ember.Component.extend({
...
-    this.get('availabilities').forEach(function (availability) {
+    this.get('records').forEach(function (appointment) {
       var event = {
-        id: availability.get('forest-chef').get('id'),
-        title: availability.get('forest-chef').get('forest-lastname'),
-        start: availability.get('forest-available_at')
+          id: appointment.get('id'),
+          title: appointment.get('forest-period'),
+          start: appointment.get('forest-startedAt'),
+          end: appointment.get('forest-finishedAt')
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: You must keep the fields for the availabilities example (`forest-available_at`, `forest-chef`,...).

File: _includes/common/views/smart-view.md [hunk 2]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
...
+      if (this.get('viewList.recordPerPage') !== 50) {
+        this.set('viewList.recordPerPage', 50);
+        this.sendAction('updateRecordPerPage');
+      }
+
+      that.set('calendarId', `${this.get('element.id')}-calendar`);
+
       Ember.$.getScript('//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js', function () {
         that.set('loaded', true);
+        debugger;
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: To Remove, my bad.

File: _includes/common/views/smart-view.md [hunk 3]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
 ```javascript
 'use strict';
 import Ember from 'ember';
-
-export default Ember.Component.extend({
+import SmartViewMixin from 'client/mixins/smart-view-mixin';
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: You should add a new line for code readability.

File: _includes/common/views/smart-view.md [hunk 4]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
...
-              timezone: 'Europe/Paris',
-              'page[size]': 100
-            };
-
-            that.get('store')
-              .query('forest_chef-availability', params)
-              .then((availabilities) => {
-                that.set('availabilities', availabilities);
-              });
+            debugger;
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: To Remove, my bad.

File: _includes/common/views/smart-view.md [hunk 5]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
...
+            console.log(that.get('collection.fields'));
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: To Remove, my bad.

File: _includes/common/views/smart-view.md [hunk 6]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
...
+            const field = that.get('collection.fields').findBy('field', 'startedAt');
+            console.log(field);
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: To Remove, my bad.

File: _includes/common/views/smart-view.md [hunk 7]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
...
-            };
-
-            that.get('store')
-              .query('forest_chef-availability', params)
-              .then((availabilities) => {
-                that.set('availabilities', availabilities);
-              });
+            debugger;
+            console.log(that.get('collection.fields'));
+            const field = that.get('collection.fields').findBy('field', 'startedAt');
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: `startedAt` to replace by `availableAt` in the context of the current example.

File: _includes/common/views/smart-view.md [hunk 8]
```diff
@@ -420,42 +420,71 @@ Having done that, your Map view is now rendered on your list view.
...
+            conditionBefore.set('field', field);
+            conditionBefore.set('operator', 'is before');
+            conditionBefore.set('value', view.end);
+            conditionBefore.set('smartView', that.get('viewList'));
+            that.set('conditionBefore', conditionBefore);
+
+            that.sendAction('addCondition', conditionAfter, true);
+            that.sendAction('addCondition', conditionBefore, true);
+
+            debugger;
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: To Remove, my bad.

File: _includes/common/views/smart-view.md [hunk 9]
```diff
@@ -471,21 +496,23 @@ export default Ember.Component.extend({
...
-    this.get('availabilities').forEach(function (availability) {
+    this.get('records').forEach(function (appointment) {
       var event = {
-        id: availability.get('forest-chef').get('id'),
-        title: availability.get('forest-chef').get('forest-lastname'),
-        start: availability.get('forest-available_at')
+          id: appointment.get('id'),
+          title: appointment.get('forest-name'),
+          start: appointment.get('forest-start_date'),
+          end: appointment.get('forest-end_date')
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: Still nope, 

```javascript
id: availability.get('forest-chef').get('id'),
title: availability.get('forest-chef').get('forest-lastname'),
start: availability.get('forest-available_at')
```
is what we expect.

File: _includes/common/views/smart-view.md [hunk 10]
```diff
@@ -420,42 +420,67 @@ Having done that, your Map view is now rendered on your list view.
...
-              timezone: 'Europe/Paris',
-              'page[size]': 100
-            };
-
-            that.get('store')
-              .query('forest_chef-availability', params)
-              .then((availabilities) => {
-                that.set('availabilities', availabilities);
-              });
+            const field = that.get('collection.fields').findBy('field', 'startedAt');
```
Comment (arnaudbesnier) [inline _includes/common/views/smart-view.md]: See my initial comment about this line.

**Comment by Tolsa** [comment]
@arnaudbesnier there was an issue when I double-tested : I was unable to scroll down the calendar view, unable to see the last line. I added an overflow: scroll to calendar which fixes it, but I am not sure this is the best way to go.

The other changes have been made.

**Comment by SeyZ** [comment]
I'm waiting for a final version on the live demo before updating the doc for mongoose & sequelize

**Comment by Tolsa** [comment]
@SeyZ as the code in Live demo has been validated, have you updated your Gist accordingly ? 
- https://gist.github.com/SeyZ/53536acf5288dc14088f54e17a1fa32e

**Pull request merged**